hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/ResizableBlockCache.java

```diff
@@ -31,4 +31,13 @@
    * @param size The max heap size.
    */
   void setMaxSize(long size);
+
+  /**
+   * Check if block type is meta or index block
+   * @param blockType block type of a given HFile block
+   * @return true if block type is non-data block
+   */
+  default boolean isMetaBlock(BlockType blockType) {
```

Comment:
isMetaBlock () is not just applicable for ResizableBlockCache.  For any BC, this is a useful stuff and so better will sit in BlockCache only?